api/kds.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package envoy.api.v2;
+
+import "api/discovery.proto";
+import "api/tls_context.proto";
+
+import "google/api/annotations.proto";
+
+service KeyDiscoveryService{
```

Comment:
Should we just name this SecretDiscoveryService? We have SDS back now because we moved SDS -> EDS.

Reply:
Seems reasonable, as long as we never require a `ServiceDiscoverySecret` ;)

Reply:
I'm fine with either... But it might be a bit confusing, considering that we just removed old SDS.

Reply:
To clarify - I need one or both of you to make a ruling on KDS vs SDS.

Reply:
Let's do SDS.